frontend/src/hooks/useGetHistoryEvents.ts

```diff
@@ -0,0 +1,37 @@
+import { EVENTS_QUERY_KEY } from '@/queryKeyStore'
+import { TokenPreview } from '@/types'
+import { wagmiConfig } from '@/wagmi.config'
+import { useQuery } from '@tanstack/react-query'
+import { getPublicClient } from '@wagmi/core'
+import { sepolia } from '@wagmi/core/chains'
+import { Log } from 'viem'
+
+export const useGetHistoryEvents = (selectedToken: TokenPreview, limit: number) => {
+  const publicClient = getPublicClient(wagmiConfig, { chainId: sepolia.id })
+
+  const { data: latestBlock } = useQuery({
+    queryKey: ['latestBlock', { tokenTitle: selectedToken.title }],
```

Comment:
The latest block is the same for both tokens since they are deployed on the same chain so we dont need to have separate query key depending on the token